bench/micro/core/fiber/chain.rb

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+class FiberChain
+    attr_reader :fiber
+    
+    def initialize(nxt)
+      @next = nxt
+      @fiber = Fiber.new do
+                  message = if @next.nil?
+                              42
+                            else 
+                              @next.resume
+                            end
+                  while msg = message
+                      Fiber.yield msg
+                  end
```

Comment:
This doesn't look right, after the first top-level resume, it's just going to yield the `msg` and not go through the chain.

Reply:
Yes, I think making message a method would correct this.

Reply:
You need to have the `@next.resume` in the loop basically.

Reply:
Fixed.